src/main/java/io/coala/jetbrains/analysis/CodeAnalysisRunner.java

```diff
@@ -149,4 +155,104 @@ public void onTextAvailable(@NotNull ProcessEvent event, @NotNull Key outputType
       }
     });
   }
+
+  private String determineFinalText(String text, CodeInspectionSeverity severity,
+      boolean isOverflow) {
+    if (!isOverflow) {
+      if (severity == CodeInspectionSeverity.ERROR || severity == CodeInspectionSeverity.DEBUG) {
+        return text.substring(7);
+      } else if (severity == CodeInspectionSeverity.WARNING) {
+        return text.substring(9);
+      }
+    }
+
+    return text;
+  }
+
+  private void printToConsole(String text, CodeInspectionSeverity prefixSeverity,
+      CodeInspectionSeverity suffixSeverity, boolean isOverflow) {
+    final String errorPrefix = "[ERROR]";
+    final String debugPrefix = "[DEBUG]";
+    final String warningPrefix = "[WARNING]";
+
+    if (!isOverflow) {
+      final String textToPrint = determineFinalText(text, prefixSeverity, isOverflow);
+
+      if (prefixSeverity == CodeInspectionSeverity.ERROR) {
+        codeAnalysisConsoleView.print(errorPrefix, prefixSeverity);
+      } else if (prefixSeverity == CodeInspectionSeverity.DEBUG) {
+        codeAnalysisConsoleView.print(debugPrefix, prefixSeverity);
+      } else if (prefixSeverity == CodeInspectionSeverity.WARNING) {
+        codeAnalysisConsoleView.print(warningPrefix, prefixSeverity);
+      } else {
+        codeAnalysisConsoleView.print("\n", CodeInspectionSeverity.INFO);
+      }
+
+      codeAnalysisConsoleView.print(textToPrint, suffixSeverity);
+    } else {
+      if (prefixSeverity == CodeInspectionSeverity.WARNING) {
+        codeAnalysisConsoleView.print("\b\b\b\b\b\b\b\b\b" + warningPrefix, prefixSeverity);
+      } else if (prefixSeverity == CodeInspectionSeverity.ERROR) {
+        codeAnalysisConsoleView.print("\b\b\b\b\b\b\b" + errorPrefix, prefixSeverity);
+      } else if (prefixSeverity == CodeInspectionSeverity.DEBUG) {
+        codeAnalysisConsoleView.print("\b\b\b\b\b\b\b" + debugPrefix, prefixSeverity);
+      } else {
+        codeAnalysisConsoleView.print("\n", CodeInspectionSeverity.INFO);
+      }
+
+      codeAnalysisConsoleView.print(text, suffixSeverity);
+    }
+  }
+
+  private void addConsolePrinterListener(@NotNull OSProcessHandler processHandler) {
+    processHandler.addProcessListener(new ProcessAdapter() {
+      String previousOutput = "";
+
+      @Override
+      public void onTextAvailable(@NotNull ProcessEvent event, @NotNull Key outputType) {
+        final String text = event.getText();
+
+        if (text == null) {
+          return;
+        }
+
+        if (outputType.equals(ProcessOutputTypes.STDERR)) {
+          final String severityCheckText =
+              text.length() >= 10 ? text.substring(0, 10).toLowerCase() : null;
+          final int previousOutputLength = previousOutput.length();
+          final String previousOutputSeverityCheckText =
+              previousOutput.length() >= 7 ? previousOutput.substring(previousOutputLength - 7)
+                  .toLowerCase() : "";
+
+          if (severityCheckText != null) {
```

Comment:
These ugly if checks are made because for some reason the `onTextAvailable` listener is fired randomly and hence in some cases instead of getting the following output
```
[DEBUG][23:15:05] Running bear coalaBear...\n
[DEBUG][23:15:05] Running bear CheckstyleBear...\n
```
line by line, it may produce
```
[DEBUG][23:15:05] Running bear coalaBear...\n[DEBUG]
[23:15:05] Running bear CheckstyleBear...\n
```

Hence these checks are done.

Reply:
We don't need such hacky code, you can just check `coala` return code and log a single statement `reaporting the status of `coala `, meanwhile, sending coala logs to a file.

Reply:
Is that possible and an option we should consider?

Reply:
The statement could be done. But what about some important errors and warnings that coala shows? For example, one such warning would be when `[Default]` section is used, or an error about a missing dependency due to which a bear might not work.

Reply:
Yeah, sending log to a file is good enough. Whenever `coala` exits with anything but `0`, just show a message asking to refer to the specific log file.

Reply:
`coala` always exits with 1 if there is some lint problem present. Wouldn't it be better to send the information to the console window I created (inside IntelliJ)? It is easier to implement this (almost done with that), and moreover, as I said logging to file isn't implemented in coala core.

Reply:
The problem I have with you current implementation is that it is too hacky and creates a very tightly coupled code package with you being the only one able to read and understand it.

1. Ignore `0`, `1` exit codes, anything other that them, show a message saying something is wrong.
2. Just redirect the log if that is possible or just get the line and write it to a file instead of you processing it yourself.

Reply:
Yes I agree this code is a bit hacky. But I think I forgot to mention that the other method which I was talking about on Gitter (running coala with `--log-json` and waiting for the process to finish completely before printing the logs) would be decoupled from the plugin and very much `elegant` (0% hacky logic). The changes I did in this PR remain discarded. I'll push that branch tomorrow.

Reply:
Thanks!